### PR TITLE
Bugfix/persistentchecks

### PR DESCRIPTION
**Pull request by tablatronix**
initial test #3641

### CODE REVIEW

**Comment by devyte** [comment]
@tablatronix will you be making the requested minor changes?

**Comment by tablatronix** [comment]
Requested minor changes?

**Comment by devyte** [comment]
Those :p just fix indentation, please. I'd like to have this ready to merge.

**Comment by tablatronix** [comment]
woah that just showed up now, weird.

File: libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp [hunk 1]
```diff
@@ -131,9 +131,11 @@ wl_status_t ESP8266WiFiSTAClass::begin(const char* ssid, const char *passphrase,
         conf.bssid_set = 0;
     }
 
-    struct station_config current_conf;
-    wifi_station_get_config(&current_conf);
-    if(sta_config_equal(current_conf, conf)) {
+    struct station_config conf_compare;
+    if(WiFi._persistent) wifi_station_get_config_default(&conf_compare);
+    else wifi_station_get_config(&conf_compare);
```
[the remark this answers is not in the document]
Comment (tablatronix) [reply libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp]: Do you mean add brackets, or is it using tabs ?

**Comment by devyte** [comment]
Thanks!

**Pull request merged**